<a href="https://colab.research.google.com/github/JulioHenri/Tecnicas-de-amostragem/blob/main/amostragem_aleatoria.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Importando bibliotecas

In [ ]:
import pandas as pd
import numpy as np

# Carregando e lendo os dados

In [ ]:
dados = pd.read_csv('HR_Engagement_Sat_Sales_UpdatedV4.0.csv') #carregando arquivo csv
dados.head() #lendo os 5 primeiros registros do dataset

ID  ... Sensor_Proximity(1-highest/10-lowest)
0   1  ...                                     9
1   2  ...                                     8
2   3  ...                                     8
3   4  ...                                     6
4   5  ...                                    10

[5 rows x 62 columns]

In [ ]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14999 entries, 0 to 14998
Data columns (total 62 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   ID                                     14999 non-null  int64  
 1   Name                                   14997 non-null  object 
 2   Department                             14999 non-null  object 
 3   GEO                                    14999 non-null  object 
 4   Role                                   14999 non-null  object 
 5   Rising_Star                            21 non-null     float64
 6   Will_Relocate                          14999 non-null  int64  
 7   Critical                               480 non-null    float64
 8   Trending Perf                          14999 non-null  int64  
 9   Talent_Level                           14999 non-null  int64  
 10  Validated_Talent_Level                 14999 non-null  int64  
 11  Pe

Podemos notar acima que há 14999 registros e 62 colunas. Também há várias atributos com valores faltantes que serão analisados posteriormente.

# Amostragem Simples

Simplesmente selecionar alguns valores do banco de dados de maneira aleatória. Nesse caso irei selecionar 40% dos meus registros.

In [ ]:
amostra = dados.sample(frac=0.40)
amostra.head()

ID  ... Sensor_Proximity(1-highest/10-lowest)
5614    5615  ...                                     9
2945    2946  ...                                     7
7682    7683  ...                                     8
13441  13442  ...                                     2
14161  14162  ...                                     4

[5 rows x 62 columns]

In [ ]:
amostra.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6000 entries, 5614 to 5219
Data columns (total 62 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   ID                                     6000 non-null   int64  
 1   Name                                   6000 non-null   object 
 2   Department                             6000 non-null   object 
 3   GEO                                    6000 non-null   object 
 4   Role                                   6000 non-null   object 
 5   Rising_Star                            11 non-null     float64
 6   Will_Relocate                          6000 non-null   int64  
 7   Critical                               196 non-null    float64
 8   Trending Perf                          6000 non-null   int64  
 9   Talent_Level                           6000 non-null   int64  
 10  Validated_Talent_Level                 6000 non-null   int64  
 11  P

Obtendo a amostra, temos 6000 registros, selecionando 40% da população de maneira aleatória. 

# Amostragem Sistemática

Os elementos da população serão colocados em uma lista e cada enésimo elemento da lista é escolhido (sistematicamente) e irá compor a amostra.

In [ ]:
seed = np.random.choice(10,1) #selecionando número aleatório de 1 a 10
seed

array([8])

In [ ]:
indices = np.arange(0, len(dados), seed)
indices

array([    0,     8,    16, ..., 14976, 14984, 14992])

No primeiro parâmetro do código acima eu seleciono em qual número o array deve se iniciar, no segundo em que número ele deve parar e por último, o intervalo com que os números devem ser escolhidos.


**Obs1**: o método 'np.arange' retorna um array (lista) de valores.

**Obs2**: len(dados) retorna a quantidade total de registros do DataFrame **dados**.

In [ ]:
amostra = dados.loc[indices]

No código acima eu localizei, dentro dos meus dados, somente os indíces que correspondem ao meu array 'indices', explicado anteriormente.

**OBS**: Índice de um DataFrame é o primeiro valor do registro da esquerda para direita, que não tem nenhuma label em cima.

In [ ]:
amostra.head()

ID  ... Sensor_Proximity(1-highest/10-lowest)
0    1  ...                                     9
7    8  ...                                     6
14  15  ...                                    10
21  22  ...                                     8
28  29  ...                                     9

[5 rows x 62 columns]

In [ ]:
amostra.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2143 entries, 0 to 14994
Data columns (total 62 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   ID                                     2143 non-null   int64  
 1   Name                                   2143 non-null   object 
 2   Department                             2143 non-null   object 
 3   GEO                                    2143 non-null   object 
 4   Role                                   2143 non-null   object 
 5   Rising_Star                            4 non-null      float64
 6   Will_Relocate                          2143 non-null   int64  
 7   Critical                               70 non-null     float64
 8   Trending Perf                          2143 non-null   int64  
 9   Talent_Level                           2143 non-null   int64  
 10  Validated_Talent_Level                 2143 non-null   int64  
 11  Per

Como podemos observar, a nossa amostra gerada tem 2143 registros, selecionados de maneira sistemática aleatoriamente.